test/integration/test-amp-skimlinks.js

```diff
@@ -0,0 +1,134 @@
+                <a id="non-merchant-link" href="https://google.com"> Test non-Merchant </a>
+            </div>
+        `,
+  };
+
+  describes.integration('Basic page', setup, env => {
+    let browser = null;
+    let clickLink = null;
+
+    beforeEach(() => {
+      // Simulated click event created by browser.click() does not trigger
+      // the browser navigation when dispatched on a link.
+      // Using MouseEvent("click") instead of Event("click") does,
+      // which is what we want in those tests.
+      clickLink = selector => {
+        const element = env.win.document.querySelector(selector);
+        if (element) {
+          const clickEvent = new MouseEvent('click', {bubbles: true});
+          element.dispatchEvent(clickEvent);
+        }
+      };
+      browser = new BrowserController(env.win);
+      return browser.waitForElementBuild('amp-skimlinks');
+
+    });
+
+    it('Should send the page impression tracking request', () => {
+      return RequestBank.withdraw('pageTrackingUrl').then(req => {
+        const regex = /^\/track\.php\?data=([^&]*)&?.*$/;
+        const match = regex.exec(req.url);
+
+        expect(match).to.have.lengthOf(2);
+        const data = JSON.parse(decodeURIComponent(match[1]));
+        expect(data.jv).to.equal(PLATFORM_NAME);
+        expect(data.pub).to.equal('123X123');
+        // nonblocking.io is the default canonical url
+        expect(data.pag).to.equal('http://nonblocking.io/');
+        expect(data.uuid).to.have.lengthOf(32);
+
+      });
+    });
+
+    it('Should send the links impression tracking request', () => {
+      return RequestBank.withdraw('linksTrackingUrl').then(req => {
+        const regex = /^\/link\?data=([^&]*)&?.*$/;
+        const match = regex.exec(req.url);
+
+        expect(match).to.have.lengthOf(2);
+        const data = JSON.parse(decodeURIComponent(match[1]));
+        expect(data.jv).to.equal(PLATFORM_NAME);
+        expect(data.pub).to.equal('123X123');
+        // nonblocking.io is the default canonical url
+        expect(data.pag).to.equal('http://nonblocking.io/');
+        expect(data.uuid).to.have.lengthOf(32);
+
+        expect(data.hae).to.equal(1);
+        expect(data.dl).to.deep.equal({
+          'https://nordstrom.com/': {count: 1, ae: 1},
+          'https://google.com/': {count: 1, ae: 0},
+        });
+      });
+    });
+
+    it('Should send NA-tracking on non-merchant link click ', () => {
+      // Give 500ms for amp-skimlinks to setup
+      return browser.wait(500).then(() => {
+        clickLink('a#non-merchant-link');
+
+        return RequestBank.withdraw('nonAffiliateTrackingUrl').then(req => {
+          const regex = /^\/\?call=track&data=([^&]*)&?.*$/;
+          const match = regex.exec(req.url);
+          expect(match).to.have.lengthOf(2);
+          const data = JSON.parse(decodeURIComponent(match[1]));
+          expect(data.url).to.equal('https://google.com/');
```

Comment:
Move the url to a constant would avoid repeating it everywhere. Although it's different from the used in the template since it has the trailing slash... 

Happy to keep it this way though, struggle between DRY and easier readability

Reply:
I don't think it would bring much to move it to a constant, not worth it.